add_connector.md

```diff
@@ -386,335 +648,582 @@ Within the `ConnectorCommon` trait, you'll find the following methods :
-            let mut header = vec![(
-                headers::CONTENT_TYPE.to_string(),
-                self.get_content_type().to_string().into(),
-            )];
-            let mut api_key = self.get_auth_header(&req.connector_auth_type)?;
-            header.append(&mut api_key);
-            Ok(header)
-        }
+        Ok(ErrorResponse {
+            status_code: res.status_code,
+            code: response
+                .code
+                .map_or(NO_ERROR_CODE.to_string(), |code| code.to_string()),
+            message: response.message.unwrap_or(NO_ERROR_MESSAGE.to_string()),
+            reason: Some(response.error),
+            attempt_status: None,
+            connector_transaction_id: None,
+        })
     }
 ```
 
-### OtherTraits
-**Payment :** This trait includes several other traits and is meant to represent the functionality related to payments.
+## `ConnectorIntegration` - The Payment Flow Orchestrator
+The `ConnectorIntegration` trait serves as the central coordinator that bridges three key files in Hyperswitch's connector architecture:
 
-**PaymentAuthorize :** This trait extends the `api::ConnectorIntegration `trait with specific types related to payment authorization.
+- **Defined in `api.rs`**  
+  [`crates/hyperswitch_interfaces/src/api.rs:150–153`](https://github.com/juspay/hyperswitch/blob/main/crates/hyperswitch_interfaces/src/api.rs#L150)  
+  Provides the standardized interface contracts for connector integration.
 
-**PaymentCapture :** This trait extends the `api::ConnectorIntegration `trait with specific types related to manual payment capture.
+- **Implemented in `mod.rs`**  
+  Each connector’s main file (`mod.rs`) implements the trait methods for specific payment flows like authorize, capture, refund, etc. You can see how the Tsys connector implements [ConnectorIntegration](https://github.com/juspay/hyperswitch/blob/main/crates/hyperswitch_connectors/src/connectors/tsys.rs#L219)
 
-**PaymentSync :** This trait extends the `api::ConnectorIntegration `trait with specific types related to payment retrieve.
+- **Uses types from `transformers.rs`**  
+  Contains PSP-specific request/response structs and `TryFrom` implementations that convert between Hyperswitch's internal `RouterData` format and the PSP's API format. This is where most connector-specific logic lives.
 
-**Refund :** This trait includes several other traits and is meant to represent the functionality related to Refunds.
+This orchestration enables seamless translation between Hyperswitch’s internal data structures and each payment service provider’s unique API requirements.
 
-**RefundExecute :** This trait extends the `api::ConnectorIntegration `trait with specific types related to refunds create.
+## Method-by-Method Breakdown
 
-**RefundSync :** This trait extends the `api::ConnectorIntegration `trait with specific types related to refunds retrieve.
+### Request/Response Flow  
```

Comment:
Here as well, we can use permalinks instead wherever source code links are included.